compose/legacy.py

```diff
@@ -86,34 +126,66 @@ def migrate_project_to_labels(project):
 def get_legacy_containers(
         client,
         project,
-        services,
-        stopped=False,
-        one_off=False):
+        services):
+
+    containers = client.containers(all=True)
 
-    containers = client.containers(all=stopped)
+    legacy_containers = []
+    legacy_one_off_containers = []
 
     for service in services:
         for container in containers:
+            if LABEL_VERSION in container['Labels']:
+                continue
+
             name = get_container_name(container)
-            if has_container(project, service, name, one_off=one_off):
-                yield Container.from_ps(client, container)
+            is_ours, is_one_off = check_name(project, service, name)
+
+            if not is_ours:
+                continue
 
+            container = Container.from_ps(client, container)
+
+            if is_one_off:
+                legacy_one_off_containers.append(container)
+            else:
+                legacy_containers.append(container)
+
+    return (legacy_containers, legacy_one_off_containers)
```

Comment:
Yeah - that was why I did it this way, but it does add a fair bit of complexity. As you say, I don't think it's worth the performance savings in the end. Will redo.